jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/steps/LambdaWorkflows.kt

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.upload.steps
+
+import com.intellij.openapi.project.Project
+import software.aws.toolkits.jetbrains.core.awsClient
+import software.aws.toolkits.jetbrains.core.credentials.AwsConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.toEnvironmentVariables
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamOptions
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamTemplateUtils
+import software.aws.toolkits.jetbrains.services.lambda.upload.CodeDetails
+import software.aws.toolkits.jetbrains.services.lambda.upload.FunctionDetails
+import software.aws.toolkits.jetbrains.utils.execution.steps.StepWorkflow
+import java.nio.file.Files
+import java.nio.file.Path
+
+fun createLambdaWorkflow(
+    project: Project,
+    codeDetails: CodeDetails,
+    buildDir: Path,
+    codeStorageLocation: String,
+    samOptions: SamOptions,
+    functionDetails: FunctionDetails
+): StepWorkflow {
+    val (dummyTemplate, dummyLogicalId) = createTemporaryTemplate(buildDir, codeDetails)
+    val packagedTemplate = buildDir.resolve("packaged-temp-template.yaml")
+    val envVars = createAwsEnvVars(project)
+
+    return StepWorkflow(
+        BuildLambda(dummyTemplate, buildDir, samOptions),
+        PackageLambda(dummyTemplate, packagedTemplate, dummyLogicalId, codeStorageLocation, envVars),
+        CreateLambda(project.awsClient(), functionDetails)
+    )
+}
+
+fun updateLambdaCodeWorkflow(
+    project: Project,
+    functionName: String,
+    codeDetails: CodeDetails,
+    buildDir: Path,
+    codeStorageLocation: String,
+    samOptions: SamOptions,
+    updatedFunctionDetails: FunctionDetails?
+): StepWorkflow {
+    val (dummyTemplate, dummyLogicalId) = createTemporaryTemplate(buildDir, codeDetails)
+    val packagedTemplate = buildDir.resolve("packaged-temp-template.yaml")
+    val envVars = createAwsEnvVars(project)
+
+    return StepWorkflow(
+        BuildLambda(dummyTemplate, buildDir, samOptions),
+        PackageLambda(dummyTemplate, packagedTemplate, dummyLogicalId, codeStorageLocation, envVars),
+        UpdateLambdaCode(project.awsClient(), functionName, updatedFunctionDetails)
+    )
+}
+
+private fun createAwsEnvVars(project: Project): Map<String, String> {
+    val connectSettings = AwsConnectionManager.getInstance(project).connectionSettings()
+        ?: throw IllegalStateException("Tried to update a lambda without valid AWS connection")
```

Comment:
Does this message get shown to the user/should it be localized?

Reply:
If it does, we have a coding error since this is gated behind our UI dialog, so not sure it is worth localizing